src/common/packets.hpp

```diff
@@ -209,4 +221,89 @@ DEFINE_PACKET_HEADER( TC_RESULT, 0xae3 );
 
 #pragma warning( pop )
 
+template <typename sessiontype> class PacketDatabase{
```

Comment:
I agree that it should be in a different file, but I just used this one for the time being.
CMake would make it easier, if we would use dynamic file listings instead of hardcoded ones.
That way CMake requires exactly the same amount of work as any other tools we use now.

Reply:
Currently, to add a new file in common, you have to modify 6 files to support all the different build systems
with #7722 , you would only have to modify one
but I'll move this discussion to that ticket